eduNEXT/eox-nelp
# And/add mt integration

## Description

This adds a new API client for MT and also adds new celery tasks that allows to update the training stage,for more details, check this [ticket](https://edunext.atlassian.net/browse/FUTUREX-675)

Tasks are connected to following django signals:

Post save BlockCompletion:  If the course doesn't have any grading policy and the user has complete the course, the tasks will make a post with the following data:
```
{
  "courseLMSId": "<course_id>",
  "userNationalId": "<user.username>",
  "applicationStageResult": 1    // PASS representation
}
```

COURSE_GRADE_NOW_PASSED: If the user has got a passed grade, the tasks will make a post with the following data:
```
{
  "courseLMSId": "<course_id>",
  "userNationalId": "<user.username>",
  "applicationStageResult": 1    // PASS representation
}
```
COURSE_GRADE_NOW_FAILED:  If the user hasn't got a passed grade, the tasks will make a post with the following data:
```
{
  "courseLMSId": "<course_id>",
  "userNationalId": "<user.username>",
  "applicationStageResult": 2    // FAIL representation
}
```
## Testing instructions
1. Set settings, you can get the api credentials in the postman collection that is shared in the [ticket](https://edunext.atlassian.net/browse/FUTUREX-675)
ACTIVATE_MT_COMPLETION_UPDATER = True
ACTIVATE_MT_TRAINING_STAGE = True
MINISTER_OF_TOURISM_USER = "<user>"
MINISTER_OF_TOURISM_PASSWORD = "<pass>"
MINISTER_OF_TOURISM_API_URL = "<url>"

2. Create two courses, one with grading policies and another without grading policies 
3. Complete both courses and check eox audit model 

### Expected result

###### PASS
![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/ee57864a-5bee-4cb0-b24b-01d208916790)

###### FAIL 
![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/898092b5-e00a-41b8-acea-02436ff452b8)


## Checklist for Merge
